### PR TITLE
feat: don't throw errors in fetch

### PR DESCRIPTION
**Pull request by jquense**

### CODE REVIEW

File: src/api/fetch.js [hunk 1]
```diff
@@ -3,9 +3,7 @@
 import FormData from 'form-data';
 import _fetch from 'node-fetch';
 
-import HttpError from './HttpError';
-
-type File = {
+export type File = {
```
Comment (taion) [inline src/api/fetch.js]: used?

File: src/api/fetch.js [hunk 2]
```diff
@@ -69,14 +65,5 @@ export default async function fetch({
       init.body = JSON.stringify(data);
     }
   }
-  const response = await _fetch(url, init);
-
-  if (!response.ok) {
```
Comment (taion) [inline src/api/fetch.js]: where do we throw this, then?

Comment (jquense) [reply src/api/fetch.js]: it should be thrown in the implementation of `request()` in HttpAPI i think. The problem here is that we don't count GET 404's as errors but that doesn't seem like a good default generally

File: src/api/fetch.js (hunk 1 above)
Comment (jquense) [reply src/api/fetch.js]: used in `RequestOptions `

**Comment by taion** [comment]
so this is breaking then?

**Comment by jquense** [comment]
actually this is a bit problematic, the `getValidationResult` method rely on something throwing. Tho maybe that method should be removed? it's super specific to our API. Alternatively i could have `fetch` take an `ok()` method that defaults to `resp => resp.ok` but allows custom logic, that seems like drilling a weird hole in there tho

**Comment by taion** [comment]
well, we can always move all the logic to `HttpApi` i suppose... possibly makes more sense, with this just serving as a wrapper for the underlying fetch, and all the actual API stuff happening outside

**Pull request merged**